### DISCUSSION TITLE
[Chapter 05] @PathVariable 과 @RequestParam

### DISCUSSION THREAD

**Discussion by coalong** [cowaine/spring-boot-for-msa#43] [Chapter 05. REST APIs with Spring MVC] [answered]
책에서 클라이언트가 서버에 **데이터를 전달하면서 요청**을 할 때 사용할 수 있는 어노테이션을 몇가지 설명한다.

`@PathVariable` 과 `@RequestParam`,  `@RequestHeader`, `@RequestBody` 을 통해 데이터를 전달하며 요청하고 그에 따른 응답을 받을 수 있다.

팀프로젝트를 진행하면서 어떤 상황에서 어떤 것을 사용해야 하는지 고민을 했는데,
 REST-API 설계를 하면서 어떤 기준으로 사용해야 할지 궁금하다!

**Comment by JoisFe** [comment, marked as answer]
사실 해당 질문이
 @bunsung92 님이 이전 단원에 질문하셨던
`적절한 REST-API 설계란 무엇일까요?` 와 같은 질문처럼 느껴집니다. (#19)
REST-API 설계를 명확하게 해두었다면
해당 명세에 맞게끔 적합한 어노테이션을 이용하여 코드를 작성하면 된다고 생각하기 때문입니다.
